### DISCUSSION TITLE
Can fastapi set different swgger js and css ‘s cdn?

### DISCUSSION THREAD

**Discussion by CaptionLittleBird** [fastapi/fastapi#11451] [Questions] [answered]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/pydantic/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
from fastapi import FastAPI, Path, Query, Depends
from typing import Annotated, Optional, Union
from routers import message_board
from utils.aioredis import register_aioredis
from settings import AIOREDIS
from fastapi.middleware.cors import CORSMiddleware
app = FastAPI(title='lookfor',
              swagger_ui_parameters={
                  "swagger_js_url": "https://cdn.staticfile.org/swagger-ui/5.1.0/swagger-ui-bundle.js",
                  'swagger_css_url': 'https://cdn.staticfile.org/swagger-ui/5.1.0/swagger-ui.css'}
              )
```

### Description

**swagger_ui_parameters** is not useful

### Operating System

Windows

### FastAPI Version

0.110.1

### Pydantic Version

2.7.0

### Python Version

3.10.2

**Comment by YuriiMotov** [comment, marked as answer]
Is this what you need?
https://fastapi.tiangolo.com/how-to/custom-docs-ui-assets/#custom-cdn-for-javascript-and-css

```
from fastapi import FastAPI
from fastapi.openapi.docs import (
    get_swagger_ui_html,
    get_swagger_ui_oauth2_redirect_html,
)

app = FastAPI(docs_url=None)


@app.get("/docs", include_in_schema=False)
async def custom_swagger_ui_html():
    return get_swagger_ui_html(
        openapi_url=app.openapi_url,
        title=app.title + " - Swagger UI",
        oauth2_redirect_url=app.swagger_ui_oauth2_redirect_url,
        swagger_js_url="https://unpkg.com/swagger-ui-dist@5.9.0/swagger-ui-bundle.js",
        swagger_css_url="https://unpkg.com/swagger-ui-dist@5.9.0/swagger-ui.css",
    )


@app.get(app.swagger_ui_oauth2_redirect_url, include_in_schema=False)
async def swagger_ui_redirect():
    return get_swagger_ui_oauth2_redirect_html()

```

**Reply by CaptionLittleBird** [reply]
thank you! it‘s works

**Comment by hasansezertasan** [comment]
You can also checkout #11158. Its a third party package that helps you to integrate OpenAPI Documentation interfaces, not only SwaggerUI and ReDoc but RapiDoc, Elements and Scalar.
